From 5d6577b Mon Sep 17 00:00:00 2001
From: build-fix-bot <bot@umicom>
Date: Thu, 10 Oct 2024 12:00:00 +0000
Subject: [PATCH 1/4] gui/app.c: make weak prototype match editor/editor.h

The weak declaration for umi_editor_save used `void` which conflicts with
the public header that declares `gboolean`. This causes a hard compile
error on GCC. Make the weak prototype match the header exactly.

--- a/src/gui/app.c
+++ b/src/gui/app.c
@@ -10,7 +10,7 @@
 #include "gui/app.h"
 #include "editor/editor.h"
 
-__attribute__((weak)) void umi_editor_save(UmiEditor *ed, GError **error);
+__attribute__((weak)) gboolean umi_editor_save(UmiEditor *ed, GError **error);
 
 /* If you also have a weak declaration for "umi_editor_save_as", ensure it
  * uses the correct signature as well:
@@ -18,6 +18,6 @@
  *   __attribute__((weak)) gboolean umi_editor_save_as(UmiEditor *ed, GError **error);
  */
 
